web/api/v1/api.go

```diff
@@ -1546,9 +1576,22 @@ func (api *API) rules(r *http.Request) apiFuncResult {
 
 		// If the rule group response has no rules, skip it - this means we filtered all the rules of this group.
 		if len(apiRuleGroup.Rules) > 0 {
+			if paginationRequest != nil && len(rgs) == int(paginationRequest.MaxRuleGroups) {
+				// We've reached the capacity of our page.
+				// We are looking ahead up to this point because this is ultimately where the presence of at least one rule
+				// group in a subsequent response is determined, hence requiring a nextToken.
```

Comment:
Hm, I don't think this comment is longer true. We break, we don't lookahead anywhere. While a bit more complex it might be useful? We can do this in separate PR -- essentially this is means extra call might be required only to learn there are not groups left and leave it empty (functionally fine, we can fix later.

Reply:
We are looking ahead. Please notice how we are only including the current rule group in the response on line 1576 if the size of `rgs` is smaller than `maxGroups`

I'm doing this so that we avoid having a bunch of empty pages in case the filters are applied.

Reply:
Thanks! The comment mislead me, but it's technically correct. "We are looking ahead up" - I somehow thought it will still do this in future, so perhaps:

```suggestion
```

And then `// We've reached the capacity of our page.` -> `// We've reached the capacity of our page plus one.` 

> I'm doing this so that we avoid having a bunch of empty pages in case we have a bunch of that filters are applied.

Yup, that goal makes sense.